rest_framework/serializers.py

```diff
@@ -502,7 +502,7 @@ def get_value(self, dictionary):
         # We override the default field access in order to support
         # lists in HTML forms.
         if html.is_html_input(dictionary):
-            return html.parse_html_list(dictionary, prefix=self.field_name)
+            return html.parse_html_list(dictionary, prefix=self.field_name) or empty
```

Comment:
How about using `None` if the field's `allow_null` is True, and using the field's `default` (or `[]`) otherwise?